Fix issue - semantic props overwritten when single prop used; update semantic to 1.3.1 (#2590)
rjsf-team/react-jsonschema-form: CHANGELOG.md

should change the heading of the (upcoming) version to include a major version bump.

-->
# v3.3.1 (upcoming)

# v3.3.0

## @rjsf/semantic-ui

> - "semantic-ui-react" updated to v1.3.1 (https://github.com/rjsf-team/react-jsonschema-form/pull/2590)

Reviewer: Thanks for the review @heath-freenome ! I think the only thing remaining is that I'd like to have a clearer description in the changelog to describe what issue exactly was fixed (rather than just linking to this PR).

Author: @epicfaace  Im not sure what else you want me to describe,

Reviewer: The changelog right now says "fixed passing [semantic theme specific props](https://react-jsonschema-form.readthedocs.io/en/latest/api-reference/themes/semantic-ui/uiSchema/), by only overiding defaults that match, with your own semantic props (https://github.com/rjsf-team/react-jsonschema-form/pull/2590)". That's confusing for a user who is seeing this changelog, and I'm not quite sure what it means either. Maybe we can add more sentences / an example to describe what it's about? I remember you said earlier that "The problem was that if you passed one specific theme prop of your own all other default props were lost."

This is actually why making an issue on GitHub Issues and linking it to this PR will make this much easier. It will then be really clear what exactly the issue is, and we can easily then convert the issue to a changelog description item. Do you think you'd be able to do that?

Author: @epicfaace thanks, I will create an issue and link it to this PR. But after this is merged in I think we should move this into core util to obtain theme specific props. Question though, should the default theme specific props be over ridden if a user passes his own. for example  these are the default props for a selectWidget 
```JS 
{
      inverted: "false",
      selection: true,
      fluid: true,
      scrolling: true,
      upward: false,
    }
```
should a user now pass 
```JS
{
  'ui:options' : {
     semantic: {
       fluid: false,
     }
   }
}
```

that select dropdown is going to look terrible because it has the lost the rest of the default props.

so should the new props be

```JS
{
      inverted: "false",
      selection: true,
      fluid: false,
      scrolling: true,
      upward: false,
    }
```
OR 
```JS
{
      fluid: false,
 }
```

So leave the default props in place unless that prop is passed exclusively by the user

> - fixed passing [semantic theme specific props](https://react-jsonschema-form.readthedocs.io/en/latest/api-reference/themes/semantic-ui/uiSchema/), by only overiding defaults that match, with your own semantic props (https://github.com/rjsf-team/react-jsonschema-form/pull/2590) related to this [issue 2619](https://github.com/rjsf-team/react-jsonschema-form/issues/2619)

Reviewer: ```suggestion
- fixed an issue where all semantic props overwritten when a single [semantic theme-specific prop](https://react-jsonschema-form.readthedocs.io/en/latest/api-reference/themes/semantic-ui/uiSchema/) is passed in ([issue 2619](https://github.com/rjsf-team/react-jsonschema-form/issues/2619)) (https://github.com/rjsf-team/react-jsonschema-form/pull/2590)
```

- Don't crash when non-object formData is passed in to a schema item with additionalProperties (https://github.com/rjsf-team/react-jsonschema-form/pull/2595)
- Upgrade jsonpointer to 5.0.0 to address security vulnerability (https://github.com/rjsf-team/react-jsonschema-form/pull/2599)

> 

Reviewer 2: Consider removing this blank space that was added unnecessarily

## @rjsf/fluent-ui
- fluent-ui: Allow value of 0 in TextWidget (https://github.com/rjsf-team/react-jsonschema-form/pull/2497)

## @rjsf/semantic-ui
- "semantic-ui-react" updated to v1.2.1 (https://github.com/rjsf-team/react-jsonschema-form/pull/2590)

> - fixed passing semantic theme specific props, by overiding defaults, with your own semantic props (https://github.com/rjsf-team/react-jsonschema-form/pull/2590)

Author: @epicfaace is there anything else
